metabase/metabase
# feat(sdk): Add collection browser

Adds the collection browser to the embedding SDK code. This PR contains:

- adding a `isDndAvailable` variable to the `app` slice of the redux store. This allows us to use the collection browser outside of the DndContext - otherwise the component is just not shown.
- Isolating the `CollectionItemsTable` component so we can use it in the SDK
- Adding some logic to the `CollectionBreadcrumbs` so we can use it for navigation



**_How to test_**

In Customer Zero, use the `CollectionBrowser` component. There's four props:
```
type CollectionBrowserProps = {
  collectionId?: CollectionId;
  onClick?: (item: CollectionItem) => void;
  pageSize?: number;
  visibleCollectionTypes?: CollectionItemModel[];
};
```

- `collectionId` is the collection ID to start browsing from. It will use `root` if nothing is provided. (we should probably talk about this)
- `onClick` emits an event when clicking an item in the collection.
- `pageSize` is pretty self-explanatory. defaults to 25
- `visibleCollectionTypes` is an array of different types to show. This includes `dashboard`, `card`, `collection`, etc. We should absolutely change the name of this - `visibleTypes` or something like that?




_**Open questions**_
- Should we have a default collection ID or can we allow the `root` default?
- Should we massage any data before emitting from `onClick`?
- What collection item types should we allow users to view from the SDK?
- How does using the breadcrumbs feel? Is it janky? Or should we use a simpler `back` button?
